src/App.js

```diff
@@ -0,0 +1,62 @@
+import React, { useState, useEffect } from 'react';
+import { isExpired, decodeToken } from 'react-jwt';
+
+import { login } from './methods/user/user';
+
+/*
+  These component could be lazily loaded, e.g.
+  const Login = React.lazy(() => import('./components/login/login'));
+  To allow for fast showing of the login form
+  (I couldn't quickly get Suspense to work with the testing library)
+*/
+import Application from './components/application/application';
+import Login from './components/login/login';
+
+function App() {
+  const [user, setUser] = useState(false);
+  const [loginError, setLoginError] = useState(false);
+
+  // Read token from local storage on first render
+  useEffect(() => {
+    const token = localStorage.getItem('user');
+    if (token) {
+      const decodedToken = decodeToken(token);
+      if (decodedToken && decodedToken.email) {
```

Comment:
I'm assuming this means returning early and negating the if statements?

Fixing this might create more code duplication as the return isn't a one liner as state is set regardless `setHasCheckedUser(true);`. I'm sure it's possible with some thought.

Reply:
Had a decent look at this, and I'm not seeing an obvious fix without duplicating code which seems to end up worse:

```
useEffect(() => {
    const token = localStorage.getItem('user');
    if (!token) {
      setHasCheckedUser(true);
      return;
    }
    const decodedToken = decodeToken(token);
    if (!decodedToken.email) {
      setHasCheckedUser(true);
      return;
    }
    setHasCheckedUser(true);
    setUser({
      email: decodedToken.email
    });
  }, []);
```